examples/platforms/emsk/radio.c

```diff
@@ -0,0 +1,668 @@
+    aIeeeEui64[4] = (NODE_ID >> 24) & 0xff;
+    aIeeeEui64[5] = (NODE_ID >> 16) & 0xff;
+    aIeeeEui64[6] = (NODE_ID >> 8) & 0xff;
+    aIeeeEui64[7] = NODE_ID & 0xff;
+}
+
+void otPlatRadioSetPanId(otInstance *aInstance, uint16_t panid)
+{
+    (void)aInstance;
+    uint8_t pan[2];
+
+    pan[0] = (uint8_t)(panid & 0xFF);
+    pan[1] = (uint8_t)(panid >> 8);
+    mrf24j40_set_pan(pan);
+}
+
+void otPlatRadioSetExtendedAddress(otInstance *aInstance, uint8_t *address)
+{
+    (void)aInstance;
+    mrf24j40_set_eui(address);
+}
+
+void otPlatRadioSetShortAddress(otInstance *aInstance, uint16_t address)
+{
+    (void)aInstance;
+    uint8_t addr[2];
+
+    addr[0] = (uint8_t)(address & 0xFF);
+    addr[1] = (uint8_t)(address >> 8);
+    mrf24j40_set_short_addr(addr);
+}
+
+void emskRadioInit(void)
+{
+    DEV_GPIO_BIT_ISR isr;
+    DEV_GPIO_INT_CFG int_cfg;
+
+    int32_t ercd;
+    uint32_t temp;
+
+    sTransmitFrame.mLength = 0;
+    sTransmitFrame.mPsdu = sTransmitPsdu;
+    sReceiveFrame.mLength = 0;
+    sReceiveFrame.mPsdu = sReceivePsdu;
+    sAckFrame.mLength = 0;
+    sAckFrame.mPsdu = sAckPsdu;
+
+    pmrf_spi_ptr = spi_get_dev(EMSK_PMRF_0_SPI_ID);
+    ercd = pmrf_spi_ptr->spi_open(DEV_MASTER_MODE, EMSK_PMRF_0_SPIFREQ);
+
+    if ((ercd != E_OK) && (ercd != E_OPNED))
+    {
+        DBG("PmodRF2 SPI open error\r\n");
+    }
+
+    pmrf_spi_ptr->spi_control(SPI_CMD_SET_CLK_MODE, CONV2VOID(EMSK_PMRF_0_SPICLKMODE));
+
+    /*MRF24J40 wakepin:output, rstpin:output, INT_PIN:input, interrupt */
+    pmrf_gpio_ptr = gpio_get_dev(EMSK_PMRF_0_GPIO_ID);
+    ercd = pmrf_gpio_ptr->gpio_open(MRF24J40_WAKE_PIN | MRF24J40_RST_PIN);
+
+    if ((ercd != E_OK) && (ercd != E_OPNED))
+    {
+        DBG("PmodRF2 CRTL port open error");
+    }
+
+    if (ercd == E_OPNED)
+    {
+        pmrf_gpio_ptr->gpio_control(GPIO_CMD_SET_BIT_DIR_OUTPUT, (void *)(MRF24J40_WAKE_PIN | MRF24J40_RST_PIN));
+        pmrf_gpio_ptr->gpio_control(GPIO_CMD_SET_BIT_DIR_INPUT, (void *)MRF24J40_INT_PIN);
+    }
+
+    pmrf_gpio_ptr->gpio_control(GPIO_CMD_DIS_BIT_INT, (void *)MRF24J40_INT_PIN);
+
+    temp = MRF24J40_INT_PIN;
+    int_cfg.int_bit_mask = temp;
+    int_cfg.int_bit_type = GPIO_INT_BITS_EDGE_TRIG(temp);
+    int_cfg.int_bit_polarity = GPIO_INT_BITS_POL_FALL_EDGE(temp);
+    int_cfg.int_bit_debounce = GPIO_INT_BITS_DIS_DEBOUNCE(temp);
+    pmrf_gpio_ptr->gpio_control(GPIO_CMD_SET_BIT_INT_CFG, (void *)(&int_cfg));
+
+    isr.int_bit_ofs = MRF24J40_INT_PIN_OFS;
+    isr.int_bit_handler = RadioIsr;
+    pmrf_gpio_ptr->gpio_control(GPIO_CMD_SET_BIT_ISR, (void *)(&isr));
+
+    /* interrupt for mrf24j40 is enabled at the end of the init process */
+    DBG("MRF24J40 Init Started\r\n");
+    mrf24j40_initialize();
+    DBG("MRF24J40 Init Finished\r\n");
+
+    pmrf_gpio_ptr->gpio_control(GPIO_CMD_ENA_BIT_INT, (void *)MRF24J40_INT_PIN);
+
+}
+
+bool otPlatRadioIsEnabled(otInstance *aInstance)
+{
+    (void)aInstance;
+    return (sState != kStateDisabled) ? true : false;
```

Comment:
You shouldn't need the inline if statement here, the comparison will return T/F on its own.